api/bash/delete-empty-repos.sh

```diff
@@ -0,0 +1,230 @@
+if [[ ${EXECUTE} != "TRUE" ]] && [[ ${EXECUTE} != "FALSE" ]]; then
+    echo "ERROR: EXECUTE was not set to a proper value."
+    echo "  Ex: bash delete-empty-repos.sh --org=MyOrganization --execute=TRUE"
+    echo ""
+    echo "Exiting script with no changes."
+    echo ""
+    exit 1
+fi
+
+if [[ ${EXECUTE} = "TRUE" ]]; then
+  echo "Searching for empty repositories within the Organization: "${ORG_NAME}
+  echo "EXECUTE was set to TRUE!!! Empty repositories will be deleted!!!"
+  echo "You have 5 seconds to cancel this script."
+  sleep 5
+else
+  echo "Searching for empty repositories within the Organization: "${ORG_NAME}
+  echo "EXECUTE was set to FALSE, no repositories will be deleted."
+fi
+
+##################################################
+# Grab JSON of all repositories for organization #
+##################################################
+echo "Getting a list of the repositories within "${ORG_NAME}
+
+REPO_RESPONSE="$(curl --request GET \
+--url ${API_ROOT}/orgs/${ORG_NAME}/repos \
+-s \
+--header "authorization: Bearer ${GITHUB_TOKEN}" \
+--header "content-type: application/json")"
+
+##########################################################################
+# Loop through every organization's repo to get repository name and size #
+##########################################################################
+echo "Generating list of empty repositories."
+echo ""
+echo "-------------------"
+echo "| Empty Repo List |"
+echo "| Org : Repo Name |"
+echo "-------------------"
+
+for repo in $(echo "${REPO_RESPONSE}" | jq -r '.[] | @base64');
+do
+  #####################################
+  # Get the info from the json object #
+  #####################################
+  get_repo_info()
+  {
+    echo ${repo} | base64 --decode | jq -r ${1}
+  }
+
+  # Get the info from the JSON object
+  REPO_NAME=$(get_repo_info '.name')
+  REPO_SIZE=$(get_repo_info '.size')
+
+  # If repository has data, size will not be zero, therefore skip.
+  if [[ ${REPO_SIZE} -ne 0 ]]; then
+    continue;
+  fi
+
+  ################################################
+  # If we are NOT deleting repository, list them #
+  ################################################
+  if [[ ${EXECUTE} = "FALSE" ]]; then
+    echo "${ORG_NAME}:${REPO_NAME}"
+
+    # Increment counter
+    EMPTY_REPO_COUNTER=$((EMPTY_REPO_COUNTER+1))
+
+  #################################################
+  # EXECUTE is TRUE, we are deleting repositories #
+  #################################################
+  elif [[ ${EXECUTE} = "TRUE" ]]; then
+      echo "${REPO_NAME} will be deleted from ${ORG_NAME}!"
+
+      ############################
+      # Call API to delete repos #
+      ############################
+      curl --request DELETE \
+         -s \
+         --url ${API_ROOT}/repos/${ORG_NAME}/${REPO_NAME} \
+         --header "authorization: Bearer ${GITHUB_TOKEN}"
+
+      echo "${REPO_NAME} was deleted from ${ORG_NAME} successfully."
```

Comment:
This may be for a further iteration but should we capture/check the return code?

Reply:
Yeah, this one has minimal error handling. Personally I'd rather open an issue for pagination and for error handling to improve this, and work on that in the future. Thoughts?